indico/modules/events/registration/client/js/form/FormErrorList.jsx

```diff
@@ -0,0 +1,101 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import React, {useMemo} from 'react';
+import {useForm} from 'react-final-form';
+import {useSelector} from 'react-redux';
+
+import {Translate} from 'indico/react/i18n';
+
+import {getFlatFieldList} from './selectors';
+import './FormErrorList.module.scss';
+
+export default function FormErrorList() {
+  const formFieldList = useSelector(getFlatFieldList);
+  const fieldLabelLookup = useMemo(() => {
+    const lookup = new Map();
+    for (const formField of formFieldList) {
+      lookup.set(formField.htmlName, {
+        label: formField.title,
+        id: `input-${formField.id}`,
+      });
+    }
+    lookup.set('captcha', {
+      label: Translate.string('Captcha'),
+      id: 'input-captcha',
+    });
+    return lookup;
+  }, [formFieldList]);
```

Comment:
I'd prefer putting this logic in a selector:

```js
export const getFieldLabelMapping = createSelector(
  getItems,
  fields => {
    const mapping = new Map();
    Object.values(fields).forEach(formField => {
      mapping.set(formField.htmlName, {
        label: formField.title,
        id: `input-${formField.id}`,
      });
    });
    mapping.set('captcha', {
      label: Translate.string('Captcha'),
      id: 'input-captcha',
    });
    return mapping;
  }
);
```

(code untested)

Reply:
Good call. Will take care of that tomorrow.